analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -204,14 +188,18 @@ private static void CheckProperty(SymbolAnalysisContext context, INamedTypeSymbo
             }
         }
 
-        private static void CheckEventOrMethod<T>(T eventOrMethod, INamedTypeSymbol containerClassSymbol,
-            SymbolAnalysisContext context, Func<T, Location> locationSelector, string memberType) where T : ISymbol
+        private static void CheckEventOrMethod<T>(T eventOrMethod,
+                                                  INamedTypeSymbol containerClassSymbol,
+                                                  SymbolAnalysisContext context,
+                                                  Func<T, Location> locationSelector,
+                                                  string memberType) where T : ISymbol
         {
-            var shadowsMethod = GetSelfAndOuterClasses(containerClassSymbol)
-                .SelectMany(c=> c.GetMembers(eventOrMethod.Name))
+            var selfAndOutterClasses = GetSelfAndOuterClasses(containerClassSymbol);
```

Comment:
I introduced the local `selfAndOutterClasses` to avoid the creation of the same list twice. To make clear, that `selfAndOutterClasses` isn't lazy, I also changed the type from `IEnumerable<T>` to `IReadOnlyList<T>`.